src/subclass/mod.rs

```diff
@@ -12,28 +12,60 @@
 //! ```rust
 //! #[macro_use]
 //! extern crate glib;
+//! #[macro_use]
+//! extern crate genum_derive;
 //! use glib::prelude::*;
 //! use glib::subclass;
 //! use glib::subclass::prelude::*;
 //!
-//! use std::cell::RefCell;
+//! use std::cell::{Cell, RefCell};
+//!
+//! #[derive(Debug, Eq, PartialEq, Clone, Copy, GEnum)]
+//! #[repr(u32)]
+//! #[genum(type_name = "SimpleObjectAnimal")]
+//! enum Animal {
+//!   Goat = 0,
+//!   #[genum(name = "The Dog")]
+//!   Dog = 1,
+//!   #[genum(name = "The Cat", nick = "chat")]
```

Comment:
Maybe should get some docs somewhere that also mention what the default values for name/nick would be and that type_name is mandatory